flocker/dockerplugin/__init__.py

```diff
@@ -5,3 +5,13 @@
 
 This may eventually be a standalone package.
 """
+
+from twisted.internet.address import UNIXAddress
+
+# Many places in both twisted.web and Klein are unhappy with listening on
+# Unix socket, e.g.  https://twistedmatrix.com/trac/ticket/5406.  The
+# Docker plugin needs to listen using a Unix socket, so "fix" that by
+# pretending we have a port number and host. Yes, I feel guilty.
+UNIXAddress.port = 0
+UNIXAddress.host = b"127.0.0.1"
+del UNIXAddress
```

Comment:
Oh, I see, the "non-docker-plugin Flocker service" case. If that ever happens then we'd have acceptance tests that fail so that's probably OK. Especially if we upstream a (better) fix.